simpeg/fields.py

```diff
@@ -5,24 +5,71 @@
 
 
 class Fields:
-    """Fancy Field Storage
+    r"""Base class for storing fields.
+
+    Fields classes are used to store the discrete field solution for a
+    corresponding simulation object; see :py:class:`SimPEG.simulation.BaseSimulation`.
+    Generally only one field solution (e.g. 'eSolution', 'phiSolution', 'bSolution') is stored.
+    However, it may be possible to extract multiple field types (e.g. 'e', 'b', 'j', 'h')
+    on the fly from the fields object. The field solution that is stored and the
+    field types that can be extracted depend on the formulation used by the associated simulation.
+    See the example below to learn how fields are extracted from fields objects.
+
+    Parameters
+    ----------
+    simulation : SimPEG.simulation.BaseSimulation
+        The simulation object used to compute the discrete field solution.
+    knownFields : dict of {key: str}, optional
+        Dictionary defining the field solutions that are stored and where
+        on the mesh they are discretized. E.g. `{'eSolution': 'E', 'bSolution': 'F'}
+        would store the `eSolution` on edges and `bSolution` on faces.
+        The ``str`` must be one of {'CC', 'N', 'E', 'F'}.
+    aliasFields : dict of {key: list}, optional
+        Set aliases to extract different field types from the field solutions that are
+        stored by the fields object. The ``key`` defines the name you would like to use
+        when extracting a given field type from the fields object. In order, the list
+        contains:
+
+        * the key for the known field solution that is used to compute the field type
+        * where the output field type lives {'CC', 'N', 'E', 'F'}
+        * the name of the method used to compute the output field.
+
+        E.g. {'b': ['eSolution', 'F', '_b']} is an alias that
+        would allow you to extract a field type ('b') that lives on mesh faces ('F')
+        from the E-field solution ('eSolution') by calling a method ('_b').
```

Comment:
```suggestion
        would allow you to extract a field type (``'b'``) that lives on mesh faces (``'F'``)
        from the E-field solution (``'eSolution'``) by calling a method (``'_b'``).
```